controllers/contacts/updateExistContact.js

```diff
@@ -0,0 +1,31 @@
+import { updateContact } from "../../service/index.js";
+import { validateUpdateContact } from "../../validator.js";
+
+export async function updateExistContact(req, res, next) {
+  const { contactId } = req.params;
+  const fields = req.body;
+  const { error } = validateUpdateContact(req.body);
+  try {
+    if (error) {
+      console.log(error);
+      return res.json({ status: 400, msg: "Missing fields" });
+    }
+    const result = await updateContact(contactId, fields);
+    if (result) {
+      return res.json({
+        status: "success",
+        code: 200,
+        data: { contact: result },
+      });
+    } else {
+      return res.status(404).json({
+        status: "error",
+        code: 404,
+        message: `Not found contact id: ${contactId}`,
+        data: "Not found",
+      });
+    }
```

Comment:
Ten kod który zaznaczyłem jest w porządku natomiast zobacz jak można go uprościć. 
```suggestion
    if (result) 
      return res.json({
        status: "success",
        code: 200,
        data: { contact: result },
      });

      return res.status(404).json({
        status: "error",
        code: 404,
        message: `Not found contact id: ${contactId}`,
        data: "Not found",
      });
```

W powyższym przykładzie usunąłem else, ponieważ kiedy robisz `if` to dajesz return. To return sprawia że w tym miejscu zakończysz operację tej funkcji, ona nie pójdzie dalej i w związku z tym kolejne return się nie wykona. Taki sposób pisania nazywa się early return. On umożliwia między innymi omijanie bloków if else. Zobacz czy ci się to może bardziej  podobać i czy tobie też nie ułatwi czytanie kodu.

Reply:
Oczywiście że ułatwia, zapomniałam o tej metodyce. Wprowadziłam ją też w innych plikach :) Dzięki za słuszną uwagę.